packages/next/src/cli/next-dev.ts

```diff
@@ -307,7 +308,30 @@ const nextDev: CliCommand = async (argv) => {
   // We do not set a default host value here to prevent breaking
   // some set-ups that rely on listening on other interfaces
   const host = args['--hostname']
-  config = await loadConfig(PHASE_DEVELOPMENT_SERVER, dir)
+
+  const { loadedEnvFiles } = loadEnvConfig(dir, true, console, false)
+
+  let envInfo: string[] = []
+  let expFeatureInfo: string[] = []
+  config = await loadConfig(
+    PHASE_DEVELOPMENT_SERVER,
+    dir,
+    undefined,
+    undefined,
+    undefined,
+    (userConfig) => {
+      const userNextConfigExperimental = getEnabledExperimentalFeatures(
+        userConfig.experimental
+      )
+      expFeatureInfo = userNextConfigExperimental.sort(
+        (a, b) => a.length - b.length
+      )
+    }
+  )
+  if (loadedEnvFiles.length > 0) {
+    envInfo = loadedEnvFiles.map((f) => f.path)
+  }
```

Comment:
Moved `envInfo` down, I generally prefer to keep the let close to where it's assigned as when the code is later moved / deleted you end up having to find if it's still used anywhere otherwise.

```suggestion
  let expFeatureInfo: string[] = []
  config = await loadConfig(
    PHASE_DEVELOPMENT_SERVER,
    dir,
    undefined,
    undefined,
    undefined,
    (userConfig) => {
      const userNextConfigExperimental = getEnabledExperimentalFeatures(
        userConfig.experimental
      )
      expFeatureInfo = userNextConfigExperimental.sort(
        (a, b) => a.length - b.length
      )
    }
  )
  
  let envInfo: string[] = []
  if (loadedEnvFiles.length > 0) {
    envInfo = loadedEnvFiles.map((f) => f.path)
  }
```